widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx

```diff
@@ -49,57 +52,30 @@ export function ConfirmWalletsModal(props: PropTypes) {
     bestRoute,
     setSelectedWallets: selectRouteWallets,
     setRouteWalletConfirmed,
-  } = useBestRouteStore();
-  const {
-    connectedWallets,
     customDestination,
     setCustomDestination,
-    selectWallets,
-  } = useWalletsStore();
+  } = useBestRouteStore();
+  const { connectedWallets, selectWallets } = useWalletsStore();
 
   const [showMoreWalletFor, setShowMoreWalletFor] = useState('');
   const [balanceWarnings, setBalanceWarnings] = useState<string[]>([]);
+  const [error, setError] = useState('');
   const [destination, setDestination] = useState(customDestination);
-  const [showCustomDestination, setShowCustomDestination] = useState(false);
+  const [showCustomDestination, setShowCustomDestination] = useState(
+    !!customDestination
+  );
   const requiredWallets = getRequiredWallets(bestRoute);
+  const customDestinationEnabled =
+    typeof config?.customDestination === 'undefined'
+      ? true
+      : config.customDestination;
 
-  const initSelectedWallets = () => {
-    const requiredChains = getRequiredChains(bestRoute);
-    const selectedWallets: Wallet[] = [];
-    requiredChains.forEach((chain) => {
-      const firstSelectedWalletWithMatchedChain = connectedWallets.find(
-        (wallet) => wallet.chain === chain && wallet.selected
-      );
-
-      if (firstSelectedWalletWithMatchedChain) {
-        selectedWallets.push({
-          walletType: firstSelectedWalletWithMatchedChain.walletType,
-          address: firstSelectedWalletWithMatchedChain.address,
-          chain: firstSelectedWalletWithMatchedChain.chain,
-        });
-      } else {
-        const firstWalletWithMatchedChain = connectedWallets.find(
-          (wallet) => wallet.chain === chain
-        );
-        if (firstWalletWithMatchedChain) {
-          selectedWallets.push({
-            walletType: firstWalletWithMatchedChain.walletType,
-            address: firstWalletWithMatchedChain.address,
-            chain: firstWalletWithMatchedChain.chain,
-          });
-        }
-      }
-    });
-
-    return selectedWallets;
-  };
-
-  const [selectableWallets, setSelectableWallets] = useState<Wallet[]>(
-    initSelectedWallets().map((item) => ({
-      walletType: item.walletType,
-      chain: item.chain,
-      address: item.address,
-    }))
+  const [selectableWallets, setSelectableWallets] = useState<ConnectedWallet[]>(
+    connectedWallets.filter(
+      (connectedWallet) =>
+        connectedWallet.selected &&
+        requiredWallets.includes(connectedWallet.chain)
```

Comment:
Please consider using 'blockchain' instead of 'chain' everywhere related to confirm pages. I mention some similar cases for this problem during this review but not all of them. please double check your PR diff at the end.

Reply:
It will be fixed in rf-886